app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -1,33 +1,141 @@
+                        it.savedInternally = true
+                    }
+                    UpsertEventResult(photosTooBig = photosDeleted, resource = Resource.Error())
+                }
+                is Resource.Success -> {
+                    result.data?.let {
+                        db.getEventDao().upsertEvent(
+                            it.toEventEntity(
+                                isDone = compressedEvent.isDone,
+                                isGoing = compressedEvent.isGoing
+                            )
+                        )
+                    }
+                    UpsertEventResult(photosTooBig = photosDeleted, resource = Resource.Success())
+                }
+            }
+        } else {
+            val result = getResourceResult {
+                api.updateEvent(
+                    eventData = MultipartBody.Part
+                        .createFormData(
+                            name = "update_event_request",
+                            value = Json.encodeToString(compressedEvent.toUpdateEventRequest())
+                        ),
+                    photoData = compressedEvent.photos.filterIsInstance<LocalPhoto>().mapIndexed {
+                        index, eventPhoto ->
+                        MultipartBody.Part
+                            .createFormData(
+                                name = "photo$index",
+                                filename = eventPhoto.key,
+                                body = eventPhoto.byteArray!!.toRequestBody()
+                            )
+                    }
+                )
+            }
+            return when (result) {
+                is Resource.Error -> {
+                    db.getEventDao().upsertModifiedEvent(
+                        ModifiedEventEntity(
+                            id = compressedEvent.id,
+                            modifiedType = ModifiedType.UPDATE
+                        )
+                    )
+                    compressedEvent.photos.filterIsInstance<LocalPhoto>().filter {
+                        !it.savedInternally
+                    }.forEach {
+                        saveLocalPhotoInternally(it)
+                        it.savedInternally = true
+                    }
+                    UpsertEventResult(photosTooBig = photosDeleted, resource = Resource.Error())
+                }
+                is Resource.Success -> {
+                    result.data?.let {
+                        db.getEventDao().upsertEvent(
+                            it.toEventEntity(
+                                isDone = compressedEvent.isDone,
+                                isGoing = compressedEvent.isGoing
+                            )
+                        )
+                    }
+                    compressedEvent.photos.filterIsInstance<LocalPhoto>().filter {
+                        it.savedInternally
+                    }.forEach {
+                        ImageStorage(context).deleteImage(it.key)
```

Comment:
You could create a global reference of your `ImageStorage`, so you don't need to recreate it on every loop iteration